File: 2187/Bits Trocados.cpp
#include <bits/stdc++.h>

using namespace std;

void solve(int n){
    int value = n;
    int aux = value;

    
    int cinq = 0, dez = 0, cinc = 0, um = 0;
    
    while (aux/50 != 0){
        cinq++;
        aux = aux - 50;
    }
    while (aux/10 != 0){
        dez++;
        aux = aux - 10;
    }
    while (aux/5 != 0){
        cinc++;
        aux = aux - 5;
    }
    while (aux/1 != 0){
        um++;
        aux = aux - 1;
    }

    cout << cinq <<" " << dez << " " << cinc << " " << um << "\n";
    cout << endl;

}


int main (){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n; cin >> n;
    int i = 1;
    while (n!= 0){
        cout << "Teste " << i << "\n";
        solve(n);
        cin >> n;
        i++;
    }
}

File: 2187/Bits Trocados.py

def solve(n):
    value = n
    cinq = 0
    dez = 0
    cinc = 0
    um = 0
    
    while int(value/50) != 0:
        cinq+=1
        value = value - 50
        
    
    while int(value/10) != 0:
        dez+=1
        value = value - 10
    
    while int(value/5) != 0:
        cinc+=1
        value = value - 5
    
    while int(value/1) != 0:
        um+=1
        value = value - 1

    
    print(str(cinq)+" "+str(dez)+" "+str(cinc)+" "+str(um))
    print()
    

n = int(input())
i = 1
while n!=0:

    print("Teste " + str(i))
    solve(n)
    n = int(input())
    i+=1
